Added eksengine to engines/__init__.py

mikezuniga opened the pull request:
- Added boto3 to setup.py requirements
- Added eks engine
- Added eks kubeconfig template
- Docs added
- TODO added

current issue related with kqueen-k8s-api methods:
```
api_1           | HTTP response headers: HTTPHeaderDict({'Content-Type': 'application/json', 'X-Content-Type-Options': 'nosniff', 'Date': 'Sun, 10 Jun 2018 10:12:10 GMT', 'Content-Length': '234'})
api_1           | HTTP response body: {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"services is forbidden: User \"system:anonymous\" cannot list services at the cluster scope","reason":"Forbidden","details":{"kind":"services"},"code":403}
api_1           |
api_1           | Traceback (most recent call last):
api_1           |   File "/code/kqueen/models.py", line 124, in status
api_1           |     'addons': kubernetes.list_services(filter_addons=True),
api_1           |   File "/code/kqueen/kubeapi.py", line 238, in list_services
api_1           |     include_uninitialized=include_uninitialized
api_1           |   File "/usr/local/lib/python3.6/site-packages/kubernetes/client/apis/core_v1_api.py", line 14358, in list_service_for_all_namespaces
api_1           |     (data) = self.list_service_for_all_namespaces_with_http_info(**kwargs)
api_1           |   File "/usr/local/lib/python3.6/site-packages/kubernetes/client/apis/core_v1_api.py", line 14455, in list_service_for_all_namespaces_with_http_info
api_1           |     collection_formats=collection_formats)
api_1           |   File "/usr/local/lib/python3.6/site-packages/kubernetes/client/api_client.py", line 321, in call_api
api_1           |     _return_http_data_only, collection_formats, _preload_content, _request_timeout)
api_1           |   File "/usr/local/lib/python3.6/site-packages/kubernetes/client/api_client.py", line 155, in __call_api
api_1           |     _request_timeout=_request_timeout)
api_1           |   File "/usr/local/lib/python3.6/site-packages/kubernetes/client/api_client.py", line 342, in request
api_1           |     headers=headers)
api_1           |   File "/usr/local/lib/python3.6/site-packages/kubernetes/client/rest.py", line 231, in GET
api_1           |     query_params=query_params)
api_1           |   File "/usr/local/lib/python3.6/site-packages/kubernetes/client/rest.py", line 222, in request
api_1           |     raise ApiException(http_resp=r)
api_1           | kubernetes.client.rest.ApiException: (403)
```
faced when try to parse k8s config body

mikezuniga committed:
Added eksengine to engines/__init__.py
Added boto3 to setup.py requirements
Added eks engine
Added eks kubeconfig template

ekhomyakova commented on kqueen/engines/eks.py:
```diff
@@ -0,0 +1,220 @@
+    'ACTIVE': config.get('CLUSTER_OK_STATE'),
+    'DELETING': config.get('CLUSTER_DEPROVISIONING_STATE'),
+    'FAILED': config.get('CLUSTER_ERROR_STATE'),
+    'UPDATED': config.get('CLUSTER_UPDATING_STATE')
```
`'UPDATING'` ?

mikezuniga replied:
for what i know the engine requires an updating state, so either we leave it or we need to put it to unknown

ekhomyakova commented on kqueen/engines/eks.py:
```diff
@@ -0,0 +1,220 @@
+                    'number': True
+                }
+            },
+            'roleArn': {
```
It's better to use snake_case everywhere

ekhomyakova commented on kqueen/engines/eks.py:
```diff
@@ -0,0 +1,220 @@
+                    'required': True
+                }
+            },
+            'subnetid': {
```
`subnet_id`

ekhomyakova commented on kqueen/engines/eks.py:
```diff
@@ -0,0 +1,220 @@
+                    'required': True
+                }
+            },
+            'securitygroupid': {
```
`security_group_id` or `securitygroup_id`

katyafervent commented on kqueen/engines/eks.py:
```diff
@@ -0,0 +1,220 @@
+            },
+            'subnetid': {
+                'type': 'text',
+                'label': 'Subnet Id',
```
Subnet/s Id - since more than one can be chosen. Ideally we need to get list of them and provide ability to pick subnets on the fly, but for now we need to parse string to the list at least (on the UI part)

katyafervent commented on kqueen/engines/eks.py:
```diff
@@ -0,0 +1,220 @@
+            aws_secret_access_key=aws_secret_key
+        )
+        try:
+            list(client.list_clusters())
```
can we used more lightweight request here? (since this request is sending for each cluster, even for list all clusters operation)
1) at least add maxResults parameter=1
2) or use another request maybe generate_presigned_url?

naumvd95 replied:
seems, No , coz afaik only list_clusters can provide valid check on all required User IAM roles/policies and etc.. I understand that its huge request, but we are locked by small api

mikezuniga commented on kqueen/engines/eks.py:
```diff
@@ -0,0 +1,220 @@
+            return {}
+        state = STATE_MAP.get(response['cluster']['status'], config.get('CLUSTER_UNKNOWN_STATE'))
+
+        key = 'cluster-{}-{}'.format(response['cluster']['name'], response['cluster']['name'])
```
one of them should be the id, but EKS uses the name as id at the moment, the only other option is to use the ARN which is extremely long arn:aws:eks:us-west-2:012345678910:cluster/prod

naumvd95 committed:
Refactor EKS

- Template structure defined
- Kubeconfig download fixed
- Cluster states fixed
- Engine status fixed
- TODO added

current issue related with kqueen-k8s-api methods:
```
api_1           | Traceback (most recent call last):
api_1           |   File "/code/kqueen/models.py", line 124, in status
api_1           |     'addons': kubernetes.list_services(filter_addons=True),
api_1           |   File "/code/kqueen/kubeapi.py", line 238, in list_services
api_1           |     include_uninitialized=include_uninitialized
api_1           |   File "/usr/local/lib/python3.6/site-packages/kubernetes/client/apis/core_v1_api.py", line 14358, in list_service_for_all_namespaces
api_1           |     (data) = self.list_service_for_all_namespaces_with_http_info(**kwargs)
api_1           |   File "/usr/local/lib/python3.6/site-packages/kubernetes/client/apis/core_v1_api.py", line 14455, in list_service_for_all_namespaces_with_http_info
api_1           |     collection_formats=collection_formats)
api_1           |   File "/usr/local/lib/python3.6/site-packages/kubernetes/client/api_client.py", line 321, in call_api
api_1           |     _return_http_data_only, collection_formats, _preload_content, _request_timeout)
api_1           |   File "/usr/local/lib/python3.6/site-packages/kubernetes/client/api_client.py", line 155, in __call_api
api_1           |     _request_timeout=_request_timeout)
api_1           |   File "/usr/local/lib/python3.6/site-packages/kubernetes/client/api_client.py", line 342, in request
api_1           |     headers=headers)
api_1           |   File "/usr/local/lib/python3.6/site-packages/kubernetes/client/rest.py", line 231, in GET
api_1           |     query_params=query_params)
api_1           |   File "/usr/local/lib/python3.6/site-packages/kubernetes/client/rest.py", line 222, in request
api_1           |     raise ApiException(http_resp=r)
api_1           | kubernetes.client.rest.ApiException: (403)
```
faced when try to parse k8s config body

naumvd95 committed:
Merge pull request #317 from Mirantis/eks-refactor

Refactor EKS

naumvd95 committed:
Docs added

- Typo fixes in EKS engine

naumvd95 committed:
Merge pull request #318 from Mirantis/kqueen-eks

Docs added

katyafervent commented on kqueen/engines/eks.py:
```diff
@@ -0,0 +1,271 @@
+        },
+        'cluster': {
+            'node_count': {
+            # TODO unsupported, need to define template k8s input
```
I suggest to comment node_count then

katyafervent commented on kqueen/engines/resources/aws/template.yaml:
```diff
@@ -0,0 +1,26 @@
+        - "-i"
+        - "<cluster-name>"
+        # - "-r"
+        # - "<role-arn>"
```
is it suppose to be commented?

naumvd95 replied:
for now yes

katyafervent reviewed (commented):
Please, add new pipfile.lock

naumvd95 committed:
Heptio Auth configuration

naumvd95 committed:
Merge pull request #321 from Mirantis/kqueen-eks

Heptio Auth configuration

naumvd95 commented:
related issue https://github.com/kubernetes-client/python/issues/514

naumvd95 reviewed (approved):
lets say I'm ok with that, but in fact I just want to hide this PR from my notifications tab :)

naumvd95 reviewed (approved):
lets say I'm ok with that, but in fact I just want to hide this PR from my notifications tab :)
